public/app/api/clients/folder/v1beta1/hooks.ts

```diff
@@ -190,6 +192,38 @@ export function useDeleteFolderMutationFacade() {
   };
 }
 
+export function useDeleteMultipleFoldersMutationFacade() {
+  const [deleteFolders] = useDeleteFoldersMutationLegacy();
+  const [deleteFolder] = useDeleteFolderMutation();
+  const dispatch = useDispatch();
+
+  if (!config.featureToggles.foldersAppPlatformAPI) {
+    return deleteFolders;
+  }
+
+  return async function deleteFolders({ folderUIDs }: { folderUIDs: string[] }) {
+    // Delete all the folders sequentially
+    // TODO error handling here
+    for (const folderUID of folderUIDs) {
+      // This also shows warning alert
+      if (await isProvisionedFolderCheck(dispatch, folderUID)) {
+        continue;
+      }
+      const result = await deleteFolder({ name: folderUID });
+      if (!result.error) {
+        // Before this was done in backend srv automatically because the old API sent a message wiht 200 request. see
+        // public/app/core/services/backend_srv.ts#L341-L361. New API does not do that so we do it here.
+        getAppEvents().publish({
+          type: AppEvents.alertSuccess.name,
+          payload: [t('folders.api.folder-deleted-success', 'Folder deleted')],
+        });
+        dispatch(refreshParents(folderUIDs));
+      }
+    }
+    return { data: undefined };
+  };
```

Comment:
Since we're not doing any improvements to the existing logic in this PR, could we have an issue created to track this, so we can get back to it later?

Reply:
I assume you mean this part you mentioned https://github.com/grafana/grafana/pull/109525#discussion_r2290612571 right?

Reply:
https://github.com/grafana/grafana/issues/110105